### PR TITLE
cephadm: Manually remove containers

### PR DESCRIPTION
**Pull request by sebastian-philipp**
This fixes:
```
Error: error creating container storage: the container name "ceph-<fsid>-mon.b" is already in use by "<container-id>". You have to remove that container to be able to reuse that name.: that name is already in use
```

Relates to https://tracker.ceph.com/issues/44990

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
```
cephadm:1857: error: Too few arguments for "rm_cmd"
cephadm:2183: error: Type signature has too few arguments
```

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -1853,6 +1853,9 @@ def deploy_daemon_units(fsid, uid, gid, daemon_type, daemon_id, c,
             install_path = find_program('install')
             f.write('{install_path} -d -m0770 -o {uid} -g {gid} /var/run/ceph/{fsid}\n'.format(install_path=install_path, fsid=fsid, uid=uid, gid=gid))
 
+        # Sometimes, adding `--rm` to a run_cmd doesn't work. Let's remove the container manually
```
Comment (mgfritch) [inline src/cephadm/cephadm:1856]: Do we know why this sometimes doesn't work? It doesn't feel like we've identified the root cause here ..

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com/swagner-2020-05-15_11:43:55-rados-wip-swagner-testing-2020-05-15-1102-distro-basic-smithi/ failed to verify this: no occurence of `"/usr/bin/ceph-mon": stat /usr/bin/ceph-mon: no such file or directory`

File: src/cephadm/cephadm (hunk 1 above)
Comment (sebastian-philipp) [reply src/cephadm/cephadm:1856]: which is correct, yes. Looks like podman doesn't honor `--rm`, if the container creation itself failed. This is a workaround for the broken `--rm` handling. 

However, this doesn't solve, why w're getting a ceph-mon: no such file or directory.

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com/swagner-2020-05-18_08:24:15-rados-wip-swagner-testing-2020-05-15-2348-distro-basic-smithi/

```
/a/swagner-2020-05-18_08:24:15-rados-wip-swagner-testing-2020-05-15-2348-distro-basic-smithi$ for d in *; do  echo  $d; grep --recursive 'stat /usr/bin/ceph-mon' $d; done
(no results)
```

**Comment by sebastian-philipp** [comment]
next opportunity: http://pulpito.ceph.com:80/swagner-2020-05-18_13:07:20-rados-wip-swagner2-testing-2020-05-18-1100-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
```
/a/swagner-2020-05-20_14:14:11-rados:cephadm-wip-swagner2-testing-2020-05-20-1008-distro-basic-smithi$ for d in *; do  echo  $d; grep --recursive 'container_linux' $d; done
(no output)
```

**Comment by sebastian-philipp** [comment]
```
/a/swagner-2020-05-27_10:06:46-rados:cephadm-wip-swagner-testing-2020-05-27-0937-distro-basic-smithi$ for d in *; do  echo  $d; grep --recursive 'container_linux' $d; done
(no output)

**Comment by sebastian-philipp** [comment]
sigh. no clue if this actually works or not.

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com/swagner-2020-05-29_10:28:21-rados:cephadm-wip-swagner-testing-2020-05-28-1651-distro-basic-smithi/

File: src/cephadm/cephadm (hunk 1 above)
Comment (tchaikov) [reply src/cephadm/cephadm:1856]: @mgfritch @sebastian-philipp we also suffered from some issues in older releases of podman and fuse-overlay but they were addressed in newer releases. see https://tracker.ceph.com/issues/44242 , https://github.com/ceph/ceph-container/commit/c44202c2eadef7262468508639644ee32ded0e07 and https://github.com/ceph/ceph-container/commit/6285bae77c3a527f872328d3e2c556eed7879475 . i think our CI (ceph-build and ceph-container) has got them addressed by installing the latest build of podman and its dependencies.

i am wondering if we should do the same in our testing harness as well?

Comment (sebastian-philipp) [reply src/cephadm/cephadm:1856]: hm. you're sure this is related?

Comment (sebastian-philipp) [reply src/cephadm/cephadm:1856]: related, sage closed https://github.com/ceph/ceph/pull/33367 a while ago without merging it.

Comment (tchaikov) [reply src/cephadm/cephadm:1856]: i am not sure. i just want to rule out whatever issues related to podman and its dependencies. if i am sure, i will put the analysis here. but i am not familiar with podman or docker at this moment, so this is what i can do.

Comment (sebastian-philipp) [reply src/cephadm/cephadm:1856]: my idea is to just merge this one and see if it helps.

**Pull request merged**